core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1428,7 +1428,11 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     */
   private def convertToOffsetMetadataOrThrow(offset: Long): LogOffsetMetadata = {
     checkLogStartOffset(offset)
-    localLog.convertToOffsetMetadataOrThrow(offset)
+    if (remoteLogEnabled() && offset < localLogStartOffset()) {
+      new LogOffsetMetadata(offset, LogOffsetMetadata.REMOTE_LOG_UNKNOWN_OFFSET)
+    } else {
+      localLog.convertToOffsetMetadataOrThrow(offset)
```

Comment:
I was thinking that we could change localLog.convertToOffsetMetadataOrThrow() such that if read() throws an exception, it just returns a message-only LogOffsetMetadata.